check_type.py

```diff
@@ -0,0 +1,60 @@
+from functools import wraps
+
+
+class InvalidTypeFunctionValue(Exception):
+    pass
+
+
+def checktype(data_type):
+    """
+    checktype
+    :param data_type:
+    :return:
+    """
+    @wraps(data_type)
+    def wrapper(func):
+        """
+        wrapper
+        :param func:
+        :return:
+        """
+        @wraps(func)
+        def wrapper2(*args):
+            """
+            wrapper 2
+            :param args:
+            :return:
+            """
+            if type(func(*args)) != data_type:
```

Comment:
You are checking if the return type is wrong, you were asked to check  the argument type.